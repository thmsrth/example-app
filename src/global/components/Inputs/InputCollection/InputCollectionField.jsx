import React from 'react';

import PropTypes from 'prop-types';
import fieldType from './types/field';

import { bemCls } from '../../../utils/ClassNameHelpers';
import { getFormInputField } from '../../../utils/FormHelpers';

const InputCollectionField = ({
  value, field, className, onChange, onBlur, optionsFromData,
}) => {
  const { hidden } = field;

  if (hidden) return null;

  return (
    <div className={`${bemCls(`InputCollection ${className}`, 'Field')} ${
      bemCls(`InputCollection ${className}`, `Field__${field.name}`)}`}
    >
      {getFormInputField(
        field,
        value,
        className,
        onChange,
        onBlur,
        optionsFromData,
        field.disabled,
      )}
    </div>
  );
};

export default InputCollectionField;

InputCollectionField.propTypes = {
  field: fieldType,
  value: PropTypes.objectOf(PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
    PropTypes.bool,
  ])),
  className: PropTypes.string,
  // eslint-disable-next-line react/forbid-prop-types
  optionsFromData: PropTypes.object,
  onChange: PropTypes.func,
  onBlur: PropTypes.func,
};

InputCollectionField.defaultProps = {
  field: {
    hidden: false,
    disabled: false,
  },
  value: null,
  className: '',
  optionsFromData: null,
  onChange: () => {},
  onBlur: () => {},
};
