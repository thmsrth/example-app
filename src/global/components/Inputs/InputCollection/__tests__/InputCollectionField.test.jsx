import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import InputCollectionField from '../InputCollectionField';

const value = {
  isValid: true,
  showWarning: false,
  value: 'firstOption',
};

const valueWithWarning = {
  isValid: true,
  showWarning: true,
  value: 'firstOption',
};

const valueWithValidationAlert = {
  isValid: false,
  showWarning: false,
  value: 'firstOption',
};

const props = {
  className: 'MyInputCollectionField',
  name: 'MyInputName',
  field: {
    name: 'mySecondField',
    title: 'My Second Field',
    type: 'select',
    required: true,
    validationAlert: 'Please select a mySecondField',
    warningAlert: 'My Second Field Warning',
    options: [
      {
        label: 'First Option',
        value: 'firstOption',
      },
      {
        label: 'Second Option',
        value: 'secondOption',
      },
    ],
  },
  value,
};

const propsWithWarning = {
  ...props,
  value: valueWithWarning,
};

const propsWithValidationAlert = {
  ...props,
  value: valueWithValidationAlert,
};

describe('InputCollectionField', () => {
  it('should render the component correctly', () => {
    const tree = shallow(<InputCollectionField {...props} />);
    expect(shallowToJson(tree)).toMatchSnapshot();
  });

  it('should render the component correctly with warnings for each items field', () => {
    const tree = shallow(<InputCollectionField {...propsWithWarning} />);
    expect(shallowToJson(tree)).toMatchSnapshot();
  });

  it('should render the component correctly with validation alert for each items field', () => {
    const tree = shallow(<InputCollectionField {...propsWithValidationAlert} />);
    expect(shallowToJson(tree)).toMatchSnapshot();
  });
});
