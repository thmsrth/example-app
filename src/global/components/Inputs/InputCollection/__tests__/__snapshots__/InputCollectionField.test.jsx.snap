// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputCollectionField should render the component correctly 1`] = `
<div
  className="InputCollectionField MyInputCollectionFieldField InputCollectionField__mySecondField MyInputCollectionFieldField__mySecondField"
>
  <InputSelectWrapper
    className="FormInput MyInputCollectionFieldInput"
    hidden={false}
    key="mySecondField"
    label="My Second Field"
    name="mySecondField"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    options={
      Array [
        Object {
          "label": "First Option",
          "value": "firstOption",
        },
        Object {
          "label": "Second Option",
          "value": "secondOption",
        },
      ]
    }
    optionsFromData={null}
    optionsFromDataKey=""
    validationKeys={
      Object {
        "onValidate": undefined,
        "onWarning": undefined,
        "pattern": undefined,
        "required": true,
        "validationAlert": "",
        "warningAlert": "",
      }
    }
    value="firstOption"
  />
</div>
`;

exports[`InputCollectionField should render the component correctly with validation alert for each items field 1`] = `
<div
  className="InputCollectionField MyInputCollectionFieldField InputCollectionField__mySecondField MyInputCollectionFieldField__mySecondField"
>
  <InputSelectWrapper
    className="FormInput MyInputCollectionFieldInput"
    hidden={false}
    key="mySecondField"
    label="My Second Field"
    name="mySecondField"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    options={
      Array [
        Object {
          "label": "First Option",
          "value": "firstOption",
        },
        Object {
          "label": "Second Option",
          "value": "secondOption",
        },
      ]
    }
    optionsFromData={null}
    optionsFromDataKey=""
    validationKeys={
      Object {
        "onValidate": undefined,
        "onWarning": undefined,
        "pattern": undefined,
        "required": true,
        "validationAlert": "Please select a mySecondField",
        "warningAlert": "",
      }
    }
    value="firstOption"
  />
</div>
`;

exports[`InputCollectionField should render the component correctly with warnings for each items field 1`] = `
<div
  className="InputCollectionField MyInputCollectionFieldField InputCollectionField__mySecondField MyInputCollectionFieldField__mySecondField"
>
  <InputSelectWrapper
    className="FormInput MyInputCollectionFieldInput"
    hidden={false}
    key="mySecondField"
    label="My Second Field"
    name="mySecondField"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    options={
      Array [
        Object {
          "label": "First Option",
          "value": "firstOption",
        },
        Object {
          "label": "Second Option",
          "value": "secondOption",
        },
      ]
    }
    optionsFromData={null}
    optionsFromDataKey=""
    validationKeys={
      Object {
        "onValidate": undefined,
        "onWarning": undefined,
        "pattern": undefined,
        "required": true,
        "validationAlert": "",
        "warningAlert": "My Second Field Warning",
      }
    }
    value="firstOption"
  />
</div>
`;
